### PR TITLE
fixes undetected url

### LINKED ISSUE #1954: url not detected in parties
### Description

Hi,
I was trying to use scancode. I ran scancode for https://raw.githubusercontent.com/rpotter12/whatsapp-play/master/setup.py

Commands used for whatsapp-play:
1. `wget https://raw.githubusercontent.com/rpotter12/whatsapp-play/master/setup.py`
2. `scancode setup.py --package --license --copyright --json-pp wplay.json`

Output: 
```
{
  "headers": [
    {
      "tool_name": "scancode-toolkit",
      "tool_version": "3.1.2",
      "options": {
        "input": [
          "setup.py"
        ],
        "--copyright": true,
        "--json-pp": "wplay.json",
        "--license": true,
        "--package": true
      },
      "notice": "Generated with ScanCode and provided on an \"AS IS\" BASIS, WITHOUT WARRANTIES\nOR CONDITIONS OF ANY KIND, either express or implied. No content created from\nScanCode should be considered or used as legal advice. Consult an Attorney\nfor any legal advice.\nScanCode is a free software code scanning tool from nexB Inc. and others.\nVisit https://github.com/nexB/scancode-toolkit/ for support and download.",
      "start_timestamp": "2020-03-06T213718.424208",
      "end_timestamp": "2020-03-06T213722.515418",
      "duration": 4.091236352920532,
      "message": null,
      "errors": [],
      "extra_data": {
        "files_count": 1
      }
    }
  ],
  "files": [
    {
      "path": "setup.py",
      "type": "file",
      "licenses": [
        {
          "key": "mit",
          "score": 100.0,
          "name": "MIT License",
          "short_name": "MIT License",
          "category": "Permissive",
          "is_exception": false,
          "owner": "MIT",
          "homepage_url": "http://opensource.org/licenses/mit-license.php",
          "text_url": "http://opensource.org/licenses/mit-license.php",
          "reference_url": "https://enterprise.dejacode.com/urn/urn:dje:license:mit",
          "spdx_license_key": "MIT",
          "spdx_url": "https://spdx.org/licenses/MIT",
          "start_line": 28,
          "end_line": 28,
          "matched_rule": {
            "identifier": "mit_30.RULE",
            "license_expression": "mit",
            "licenses": [
              "mit"
            ],
            "is_license_text": false,
            "is_license_notice": false,
            "is_license_reference": false,
            "is_license_tag": true,
            "matcher": "2-aho",
            "rule_length": 2,
            "matched_length": 2,
            "match_coverage": 100.0,
            "rule_relevance": 100.0
          }
        },
        {
          "key": "mit",
          "score": 27.0,
          "name": "MIT License",
          "short_name": "MIT License",
          "category": "Permissive",
          "is_exception": false,
          "owner": "MIT",
          "homepage_url": "http://opensource.org/licenses/mit-license.php",
          "text_url": "http://opensource.org/licenses/mit-license.php",
          "reference_url": "https://enterprise.dejacode.com/urn/urn:dje:license:mit",
          "spdx_license_key": "MIT",
          "spdx_url": "https://spdx.org/licenses/MIT",
          "start_line": 49,
          "end_line": 49,
          "matched_rule": {
            "identifier": "pypi_mit_license.RULE",
            "license_expression": "mit",
            "licenses": [
              "mit"
            ],
            "is_license_text": false,
            "is_license_notice": false,
            "is_license_reference": false,
            "is_license_tag": true,
            "matcher": "2-aho",
            "rule_length": 5,
            "matched_length": 5,
            "match_coverage": 100.0,
            "rule_relevance": 27
          }
        }
      ],
      "license_expressions": [
        "mit",
        "mit"
      ],
      "copyrights": [],
      "holders": [],
      "authors": [
        {
          "value": "Rohit Potter, Alexandre",
          "start_line": 26,
          "end_line": 29
        }
      ],
      "packages": [
        {
          "type": "pypi",
          "namespace": null,
          "name": "wplay",
          "version": "5.0.2",
          "qualifiers": {},
          "subpath": null,
          "primary_language": "Python",
          "description": "command line software to play with your WhatsApp",
          "release_date": null,
          "parties": [
            {
              "type": "person",
              "role": "author",
              "name": "Rohit Potter, Alexandre Calil",
              "email": "rohitpotter12@gmail.com, alexandrecalilmf@gmail.com",
              "url": null
            }
          ],
          "keywords": [
            "Programming Language :: Python",
            "Programming Language :: Python :: 3",
            "Programming Language :: Python :: 3.6",
            "Programming Language :: Python :: 3.7",
            "Programming Language :: Python :: 3.8",
            "Programming Language :: Python :: 3 :: Only",
            "Operating System :: OS Independent"
          ],
          "homepage_url": "https://github.com/rpotter12/whatsapp-play/",
          "download_url": null,
          "size": null,
          "sha1": null,
          "md5": null,
          "sha256": null,
          "sha512": null,
          "bug_tracking_url": null,
          "code_view_url": null,
          "vcs_url": null,
          "copyright": null,
          "license_expression": "mit",
          "declared_license": {
            "license": "MIT",
            "classifiers": [
              "License :: OSI Approved :: MIT License"
            ]
          },
          "notice_text": null,
          "root_path": null,
          "dependencies": [],
          "contains_source_code": null,
          "source_packages": [],
          "purl": "pkg:pypi/wplay@5.0.2",
          "repository_homepage_url": null,
          "repository_download_url": null,
          "api_data_url": null
        }
      ],
      "scan_errors": []
    }
  ]
}

```
***In this url is shown null but the original file contains the url***

### System configuration

* What OS are you running on? Linux
* What version of scancode-toolkit was used to generate the scan file? `ScanCode version 3.1.2`
* What installation method was used to install/run scancode? `pip`

---
If this is a bug, then I would like to solve this bug :)

### PR DESCRIPTION
**Pull request by rpotter12**
Fixes #1954 

added homepage_url in pypi.py which helps in detecting the url.

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

**Comment by rpotter12** [comment]
@pombredanne Please review this :)

File: setup.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 from setuptools import find_packages
 from setuptools import setup
 
-version = '3.1.2'
+version = '3.1.3'
```
Comment (pombredanne) [inline setup.py]: No need to update the version in your PR.

Comment (pombredanne) [reply setup.py:19]: Ping ^

File: src/packagedcode/pypi.py [hunk 2]
```diff
@@ -270,13 +270,15 @@ def parse_setup_py(location):
...
+    homepage_url = setup_args.get('url')
     if author:
         parties.append(
             models.Party(
                 type=models.party_person,
                 name=author,
                 email=author_email,
-                role='author'
+                role='author',
+                url=homepage_url
```
Comment (pombredanne) [inline src/packagedcode/pypi.py:281]: Can you add a test for this?

**Comment by rpotter12** [comment]
@pombredanne please review the PR again. I have added the required tests. :)

**Comment by rpotter12** [comment]
Ping :)

File: setup.py (hunk 1 above)
Comment (pombredanne) [reply setup.py]: Can you remove that version update?

**Comment by pombredanne** [comment]
All good... merging now! thank you!

**Comment by pombredanne** [comment]
Small note: we need to improve the commit messages in the future ;)

**Pull request merged**